### ISSUE TITLE
Blog categories and related posts compromising search results - how exclude?

### ISSUE THREAD

**Issue by CrispinF** [cloudscribe/cloudscribe.SimpleContent#442]
I have a blog in which individual posts are created for musical works. https://peter-tranchell.uk/musical-works/blog
I use categories to say what instruments/groups each work is for, e.g. "bassoon", "choir" and what kind of work it is. The problem is, Bing and Google see all these categories on every post, so individual works get associated with irrelevant terms and appear incorrectly in search results.
Example, google "tranchell bassoon" and we get "Waltz and Calypso" first (which is good) but then "Magnificat upon Tone 8" (which for choir, and so bad). Similar situation with Bing.

I use an embedded "Bing Custom Search Hosted UI" component, which works really nicely given cloudscribe has no in-built search, but obviously it suffers exactly the same problem. That is here: https://peter-tranchell.uk/search 

Related Posts has the same effect, again presenting irrelevant content for indexing and leading to incorrect results.

I know both of these are optional, and this problem could be avoided by simply switching off related posts and not using categories, but they are really useful in other ways.

Solutions?  

I know things can be fully hidden by putting them in iframes loaded from pages that are blocked by robots.txt, but that feels really horrible.

I'm seeing uncertain information about wrapping the categories and related posts sections in "" - it seems this will only work for Google Search Appliance, rather than googlebot in general, and I assume it wouldn't work for Bing/Yahoo and others. 

And this might work for Yahoo (and even Bing?) 
`<div class="robots-nocontent">Yahoo bots won't index this.</div>`

Any thoughts on this, both for my particular site and for cloudscribe in general? It must affect all blogs to some extent, so I'm really surprised I'm not finding solutions. 

Relevant discussions:
https://stackoverflow.com/questions/15685205/noindex-tag-for-google/15718255#15718255  
https://webmasters.stackexchange.com/questions/16390/preventing-robots-from-crawling-specific-part-of-a-page

**Comment by joeaudette** [comment]
I'm not sure I see a problem here. The most relevant post is found first by google and the other items returned are still about "tranchell". Seems to me your site is getting good coverage by google and doing well in search results for its content. (and that even though you don't seem to have populated meta descriptions)

What is the actual problem and why are you sure it is caused by having a category list?

**Comment by joeaudette** [comment]
and now this github issue is found as the 4th result!

**Comment by joeaudette** [comment]
more related issues:

https://meta.stackexchange.com/questions/14898/how-do-i-make-search-return-results-that-actually-contain-all-the-search-terms

https://superuser.com/questions/19335/force-google-search-to-require-terms-by-default

bottom line I think is we don't have much control to shape the search results from google or bing.

You can override views and try experiments like robots-nocontent. At the moment I'm not seeing any compelling need for change in simple content default views.

**Comment by CrispinF** [comment]
Good point - if we're simply trying to bring traffic to our site, then you're right there's no big problem.  But for the site search (using the Bing Custom Search), users have reported that it's "broken" when searching for "bassoon" finds things that are nothing to do with bassoons. And it's definitely the categories and related posts that are causing this.
I'll maybe try overriding these views and see if it makes any difference!
I guess another way of looking at this is... Search Engines are not the best tools for intra-site search. Any progress on a cloudscribe search mechanism? I remember a few months ago you were considering Elasticsearch, Algolia, or possibly full-text index searches against the database.

**Comment by joeaudette** [comment]
I would say the issue is not SEO, SEO is working fine, its just that you want more control to exclude things from search results, so maybe SUE (Search User Experience) is a better way to think of this.

One experiment you can try is to add rel=nofollow on the category links, though google generally says not to worry about that for internal links, it might be worth a try given your desired outcome. 
Another idea for experiment is to use nav element around the ul, not sure if that would be seen differently than a regular list of links but maybe.

Sorry but no, search is a hard problem and I have a large list of other problems to solve first. I'm feeling very behind on my to do list.

**Comment by CrispinF** [comment]
Agree to all that, and yes will also try those suggestions.

Quick q on overriding... if I drop a copy of CategoryListPartial.cshtml from here:

https://github.com/cloudscribe/cloudscribe.SimpleContent/blob/master/src/cloudscribe.SimpleContent.CompiledViews.Bootstrap4/Views/Blog/CategoryListPartial.cshtml 

into my Views\Blog folder, I get this error on build:
Severity	Code	Description
Error	CS1955	Non-invocable member 'KeyValuePair<string, int>.Key' cannot be used like a method.

Can you give me a pointer?

**Comment by joeaudette** [comment]
If you copied and pasted the content of the view into a new view in VS it likely messed up the formatting of the () and thinks it is a method invocation, ie it lost some needed white space. Compare yours carefully to the original and make sure you have white space in the same places.

**Comment by CrispinF** [comment]
Ha - yes, that was it, thanks!

**Comment by CrispinF** [comment]
@joeaudette I've wrapped the categories and related posts in <nav> and will report back in a few days whether this has made any difference.

**Comment by CrispinF** [comment]
@joeaudette I'm failing to override ArchiveListPartial.cshtml on the live site.
In VS my override works fine, and the archive links are within a nav element. But when I publish and deploy, the live site seems to ignore this. eg see https://peter-tranchell.uk/musical-works/blog 
There were no errors on building/publishing the site from VS.
My overridden view is here: 
https://github.com/CrispinF/cloudscribe_PeterTranchell/blob/master/cloudscribe_PeterTranchell/Views/Blog/ArchiveListPartial.cshtml 
Is there any obvious reason an override would work in VS but not in deployed site?

**Comment by joeaudette** [comment]
@CrispinF looks like VS has junked up your .csproj file and told it to remove that view, among other things. Bunch of junk in there that should be removed. I would try that, recompile, redeploy.

**Comment by CrispinF** [comment]
@joeaudette whoa yes I see.  I'm not really confident about what's going on in this file and don't want to break everything... I'll refer to a vanilla version and take it from there.
thanks!

**Comment by CrispinF** [comment]
I'm making notes about the effects of this on search here https://github.com/CrispinF/cloudscribe_PeterTranchell/issues/6

**Comment by joeaudette** [comment]
This coming change from google may reduce extra search results: https://www.engadget.com/2019/06/07/google-search-top-results-update/

**Comment by CrispinF** [comment]
It's not hard proof, but results in Google and the internal Microsoft Search are now better after wrapping the irrelevant lists in "nav".  The top results for "Tranchell bassoon" are now posts that really do feature bassoon, rather than ones where bassoon was simply a word in the categories list.
So, given this is also semantically correct, and good practice for significant groups of navigation links,, maybe it would be good to include these in the default views. The "aria-labelledby attribute" helps differentiate the different kinds of nav.

**Comment by joeaudette** [comment]
I've changed the categories and archives to use nav instead of section, and I've wrapped the related posts, recent posts, and featured posts in nav element as well. I've shipped an updated nuget for bootstrap4 https://www.nuget.org/packages/cloudscribe.SimpleContent.CompiledViews.Bootstrap4/4.0.31

I looked at your local views and noticed incorrect use of aria-labelledby, it should be the id of the heading that labels the section not literal label content.

I'm going to close this as nothing further to do.
